statistics/index.go

```diff
@@ -127,23 +127,30 @@ func (idx *Index) TotalRowCount() float64 {
 	return idx.Histogram.TotalRowCount()
 }
 
-// IsInvalid checks if this index is invalid.
-func (idx *Index) IsInvalid(sctx sessionctx.Context, collPseudo bool) (res bool) {
-	idx.CheckStats()
+// IndexStatsIsInvalid checks whether the index has valid stats or not.
+func IndexStatsIsInvalid(idxStats *Index, sctx sessionctx.Context, coll *HistColl, cid int64) (res bool) {
+	// When we are using stats from PseudoTable(), all column/index ID will be -1.
+	if ((idxStats != nil && idxStats.IsFullLoad()) || coll.PhysicalID <= 0) && !coll.CanNotTriggerLoad {
```

Comment:
I think this check is wrong.

Reply:
And can we have a unified checking in this method instead of two different points of trigger loading?
Comments are also needed.

Reply:
```
	// When we are using stats from PseudoTable(), all column/index ID will be -1.
	if idx.IsFullLoad() || idx.PhysicalID <= 0 {
		return
	}
	HistogramNeededItems.insert(model.TableItemID{TableID: idx.PhysicalID, ID: idx.Info.ID, IsIndex: true})
```
It's copied from the `CheckStats`. But yes, we can redesign it. I just want to make the review more easier by copying it instead of fully refactoring it(